src/components/YearlyDataTab.tsx

```diff
@@ -2,22 +2,33 @@ import * as React from "react";
 import CopyButton from "./CopyButton";
 import { Button } from "library-simplified-reusable-components";
 import { LibraryData } from "../interfaces";
+import StatsInnerList from "./StatsInnerList";
+import DropdownButton from "./DropdownButton";
+import { getPercentage } from "../utils/sharedFunctions";
 
 export interface YearlyDataTabProps {
   data: {[key: string]: LibraryData[]};
 }
 
 export interface YearlyDataTabState {
   styled: boolean;
+  yearsToShow: {[key: number]: boolean};
 }
 
 export default class YearlyDataTab extends React.Component<YearlyDataTabProps, YearlyDataTabState> {
   private dataRef = React.createRef<HTMLUListElement>();
   constructor(props: YearlyDataTabProps) {
     super(props);
-    this.state = { styled: true };
+    this.state = { styled: true, yearsToShow: {} };
     this.toggleFormatting = this.toggleFormatting.bind(this);
   }
+  async componentWillReceiveProps() {
+    await Object.values(this.props.data).some(x => x.length > 0);
```

Comment:
Object.values() isn't asynchronous, and I'm not sure what this is doing.

Reply:
No--getting all the values for `this.props.data` is asynchronous.  We're waiting for there to be actual libraries in the arrays, not just `{production: [], testing: [], cancelled: []}`.  This is how the dropdown knows what years we're dealing with, so it knows what to make the hide/show options for.  (This was the only way I could get that to work; I tried a bunch of things.)